unlock-app/src/components/interface/checkout/alpha/Checkout/index.tsx

```diff
@@ -30,9 +36,42 @@ export function Checkout({ paywallConfig, injectedProvider }: Props) {
     paywallConfig,
     state.value as CheckoutPage
   )
+  const { messageToSign, mint, lock } = state.context
 
-  const onClose = () => {
-    // TODO
+  useEffect(() => {
+    if (
+      mint &&
+      mint.transactionHash &&
+      mint.status === 'FINISHED' &&
+      communication
```

Comment:
can we change it with a more "speaking" var?

Reply:
🤔 I don't understand what that means. 

It is meant to be string enum of various states like `ERROR`, `PROCESSING`, `FINISHED`.

Reply:
will do